### PR TITLE
fix: snap points scroll calculations

### PR DESCRIPTION
**Pull request by kirillzyusko**
## 📜 Description

Fixed corner cases for `scrollOutput` function, added tests, updated docs and example apps.

## 💡 Motivation and Context

This is a slight revision of https://github.com/kirillzyusko/react-native-keyboard-controller/pull/452

In this PR I fixed some bugs, unified some functions, updated docs, etc.

### JS

- fixed an issue with returning absolute value from `scrollOutput` from `snapPoints`;
- renamed `scrollOutput` -> `scrollDistanceWithRespectToSnapPoints`;
- refactored function `scrollDistanceWithRespectToSnapPoints` to make less computations in cycle and make a signature of function simpler (now this function always returns absolute scroll position);
- added unit-tests;
- ported example to FabricExample;

### Docs

- added info about `snapToOffsets` and how `KeyboardAwareScrollView` handles that;

## 🤔 How Has This Been Tested?

Tested locally on iPhone 15 Pro.

Tested on CI.

## 📝 Checklist

- [x] CI successfully passed
- [x] I added new mocks and corresponding unit-tests if library API was changed

### CODE REVIEW

File: docs/versioned_docs/version-1.12.0/api/components/keyboard-aware-scroll-view.mdx [hunk 1]
```diff
@@ -73,6 +73,10 @@ The distance between keyboard and focused `TextInput` when keyboard is shown. De
 This property is equivalent to [extraHeight](https://github.com/APSL/react-native-keyboard-aware-scroll-view/tree/9eee405f7b3e261faf86a0fc8e495288d91c853e?tab=readme-ov-file#props) from original [react-native-keyboard-aware-scroll-view](https://github.com/APSL/react-native-keyboard-aware-scroll-view) package.
 :::
 
+:::tip
+If you specified `snapOffsets` to your `ScrollView` then `KeyboardAwareScrollView` will automatically respect to this values and actual `bottomOffset` may be bigger (but focused input will be always above the keyboard anyway).
```
Comment (kirillzyusko) [inline docs/versioned_docs/version-1.12.0/api/components/keyboard-aware-scroll-view.mdx:77]: these values

File: example/src/screens/Examples/AwareScrollView/index.tsx [hunk 2]
```diff
@@ -23,7 +23,6 @@ export default function AwareScrollView({ navigation }: Props) {
   const [disableScrollOnKeyboardHide, setDisableScrollOnKeyboardHide] =
     useState(false);
   const [enabled, setEnabled] = useState(true);
-
```
Comment (kirillzyusko) [inline example/src/screens/Examples/AwareScrollView/index.tsx:26]: TODO: port this example app to Fabric architecture

**Comment by kgni** [comment]
@kirillzyusko sorry for the late response. Just reinstalled my app pointing to this PR, and everything seems to be working as expected!

**Comment by kgni** [comment]
@kirillzyusko seems to be some small issues with the scrolling being stuck now. This was not happening before.
Also it feels like performance decreased on the UI thread, I was getting 60 frames before hmm

video: https://share.cleanshot.com/tQWgl0KS
